### PR TITLE
IDN M3

### PR DESCRIPTION
**Pull request by driemworks**
- moves pallets, client, and primitives to a new repo
- adds more robust tests
- introduces the randomness beacon pallet

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,7 @@ target/
 
 # Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
 # More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
+# Cargo.lock
```
Comment (juangirini) [inline .gitignore:8]:
```suggestion
```

File: build_chainspec.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # build a dev chainspec
 #cargo build --release
```
Comment (juangirini) [inline build_chainspec.sh:2]:
```suggestion
```

File: build_chainspec.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
-./target/release/node build-spec --dev > etfDevSpec.json
-./target/release/node build-spec --chain=etfDevSpec.json --raw > local_chainspec.json
+./target/debug/node build-spec --dev > etfDevSpec.json
+./target/debug/node build-spec --chain=etfDevSpec.json --raw > local_chainspec.json
 #rm etfDevSpec.json
```
Comment (juangirini) [inline build_chainspec.sh:5]:
```suggestion
```

File: compose.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  #     test: ["CMD", "curl", "-f", "ws://localhost:9944"]
+  #     interval: 10s
+  #     timeout: 5s
+  #     retries: 5
+  #   networks:
+  #     testing_net:
+  #       ipv4_address: 172.28.1.2
+  #       aliases:
+  #         - bob
+
```
Comment (juangirini) [inline compose.yml]:
```suggestion
```

File: compose.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+version: '3.7'
```
Comment (juangirini) [inline compose.yml]: shouldn't this file be called `docker-compose.yml`?

File: node/Cargo.toml [hunk 6]
```diff
@@ -50,98 +50,105 @@ log = { workspace = true, default-features = true }
...
-sp-consensus = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sp-transaction-storage-proof = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sp-io = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sp-mixnet = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sp-mmr-primitives = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sp-statement-store = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
+sp-authority-discovery = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sp-consensus-aura = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sp-consensus-babe = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+# beefy-primitives = { package = "sp-consensus-beefy-etf", path = "../../pallets/primitives/consensus/beefy-etf", features = ["bls-experimental"]}
```
Comment (juangirini) [inline node/Cargo.toml:56]:
```suggestion
```

File: node/Cargo.toml [hunk 7]
```diff
@@ -50,98 +50,105 @@ log = { workspace = true, default-features = true }
...
+sc-transaction-pool-api = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-statement-store = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-network = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-network-common = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-network-sync = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-network-statement = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-consensus-slots = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-consensus-aura = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+sc-consensus-babe = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
 # sc-consensus-babe = { path = "../client/consensus/babe" }
```
Comment (juangirini) [inline node/Cargo.toml:91]:
```suggestion
```

File: node/Cargo.toml [hunk 8]
```diff
@@ -50,98 +50,105 @@ log = { workspace = true, default-features = true }
...
-sc-telemetry = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-executor = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-authority-discovery = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-mixnet = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-sync-state-rpc = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-sysinfo = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-storage-monitor = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-offchain = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
+# beefy = { package = "sc-consensus-beefy-etf-etf", git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, features = ["bls-experimental"] }
+# beefy = { package = "sc-consensus-beefy-etf", path = "../../pallets/client/consensus/beefy-etf", features = ["bls-experimental"] }
```
Comment (juangirini) [inline node/Cargo.toml:93]:
```suggestion
```

File: compose.yml (hunk 5 above)
Comment (driemworks) [reply compose.yml]: it works both ways

File: node/Cargo.toml [hunk 9]
```diff
@@ -50,98 +50,105 @@ log = { workspace = true, default-features = true }
...
 
 # CLI-specific dependencies
-sc-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
-frame-benchmarking-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
-node-inspect = { package = "staging-node-inspect", git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
-try-runtime-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
+sc-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+frame-benchmarking-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+node-inspect = { package = "staging-node-inspect", git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+# try-runtime-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
```
Comment (juangirini) [inline node/Cargo.toml:138]:
```suggestion
```

File: node/Cargo.toml [hunk 10]
```diff
@@ -50,98 +50,105 @@ log = { workspace = true, default-features = true }
...
+frame-metadata-hash-extension = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
 
 # node-specific dependencies
-kitchensink-runtime = { path = "../runtime" }
+node-template-runtime = { path = "../runtime" }
 node-rpc = { path = "./rpc" }
-node-primitives = { path = "./primitives" }
-sp-ark-bls12-377 = { git = "https://github.com/paritytech/sp-curves.git" }
+node-primitives = { path = "./primitives" } 	
+sp-ark-bls12-381 = { git = "https://github.com/paritytech/sp-curves.git" }
```
Comment (juangirini) [inline node/Cargo.toml:125]: Let's use the crates.io version
```suggestion
sp-ark-bls12-381 = "0.4.1"
```

File: node/Cargo.toml [hunk 11]
```diff
@@ -153,42 +160,43 @@ criterion = { version = "0.4.0", features = ["async_tokio"] }
...
-pallet-timestamp = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-substrate-cli-test-utils = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
+substrate-rpc-client = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+pallet-timestamp = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+substrate-cli-test-utils = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
 
 wat = "1.0"
-frame-support = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
+frame-support = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
+# frame-metadata-hash-extension = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true }
```
Comment (juangirini) [inline node/Cargo.toml:169]:
```suggestion
```

File: node/Cargo.toml [hunk 12]
```diff
@@ -153,42 +160,43 @@ criterion = { version = "0.4.0", features = ["async_tokio"] }
...
-try-runtime-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
-sc-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev", optional = true }
-pallet-balances = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
-sc-storage-monitor = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "tony/dev" }
+node-inspect = { package = "staging-node-inspect", git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+frame-benchmarking-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+substrate-build-script-utils = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+# substrate-frame-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
+# try-runtime-cli = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = true, optional = true }
```
Comment (juangirini) [inline node/Cargo.toml:196]:
```suggestion
```

File: node/Cargo.toml [hunk 13]
```diff
@@ -200,15 +208,15 @@ cli = [
 	"sc-cli",
 	"sc-service/rocksdb",
 	"substrate-build-script-utils",
-	"substrate-frame-cli",
-	"try-runtime-cli",
+	# "substrate-frame-cli",
+	# "try-runtime-cli",
```
Comment (juangirini) [inline node/Cargo.toml:212]:
```suggestion
```

File: node/Cargo.toml [hunk 14]
```diff
@@ -245,7 +253,7 @@ try-runtime = [
 	"pallet-treasury/try-runtime",
 	"sp-runtime/try-runtime",
 	# "substrate-cli-test-utils/try-runtime",
-	"try-runtime-cli/try-runtime",
+	# "try-runtime-cli/try-runtime",
```
Comment (juangirini) [inline node/Cargo.toml:256]:
```suggestion
```

File: node/Cargo.toml [hunk 15]
```diff
@@ -245,7 +253,7 @@ try-runtime = [
 	"pallet-treasury/try-runtime",
 	"sp-runtime/try-runtime",
 	# "substrate-cli-test-utils/try-runtime",
```
Comment (juangirini) [inline node/Cargo.toml:255]:
```suggestion
```

File: runtime/Cargo.toml [hunk 16]
```diff
@@ -1,13 +1,12 @@
 [package]
-name = "kitchensink-runtime"
-version = "3.0.0-dev"
+name = "node-template-runtime"
```
Comment (juangirini) [inline runtime/Cargo.toml:2]: This should be something like "etf-runtime"

**Comment by driemworks** [comment]
I am merging it as is for the time being - will start to create backlog items to address these cleanup items.

**Pull request merged**